contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -125,24 +137,338 @@ contract OETHVaultCore is VaultCore {
+        // and checking it's queued amount is > the queue's claimable amount.
+
+        // Stream any harvested rewards (WETH) that are available to the Vault
+        IDripper(dripper).collect();
+
+        // Add any WETH from the Dripper to the withdrawal queue
+        _addWithdrawalQueueLiquidity();
+
+        amounts = new uint256[](_requestIds.length);
+        for (uint256 i = 0; i < _requestIds.length; ++i) {
+            amounts[i] = _claimWithdrawal(_requestIds[i]);
+            totalAmount += amounts[i];
+        }
+
+        // transfer all the claimed WETH from the vault to the withdrawer
+        IERC20(weth).safeTransfer(msg.sender, totalAmount);
+
+        // Prevent insolvency
+        _postRedeem(totalAmount);
+    }
+
+    function _claimWithdrawal(uint256 requestId)
+        internal
+        returns (uint256 amount)
+    {
+        // Load the structs from storage into memory
+        WithdrawalRequest memory request = withdrawalRequests[requestId];
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+
+        require(
+            request.timestamp + CLAIM_DELAY <= block.timestamp,
+            "Claim delay not met"
+        );
+        // If there isn't enough reserved liquidity in the queue to claim
+        require(request.queued <= queue.claimable, "Queue pending liquidity");
+        require(request.withdrawer == msg.sender, "Not requester");
+        require(request.claimed == false, "Already claimed");
+
+        // Store the request as claimed
+        withdrawalRequests[requestId].claimed = true;
+        // Store the updated claimed amount
+        withdrawalQueueMetadata.claimed = queue.claimed + request.amount;
+
+        emit WithdrawalClaimed(msg.sender, requestId, request.amount);
+
+        return request.amount;
+    }
+
+    /// @notice Collects harvested rewards from the Dripper as WETH then
+    /// adds WETH to the withdrawal queue if there is a funding shortfall.
+    /// @dev is called from the Native Staking strategy when validator withdrawals are processed.
+    /// It also called before any WETH is allocated to a strategy.
+    function addWithdrawalQueueLiquidity() external {
+        // Stream any harvested rewards (WETH) that are available to the Vault
+        IDripper(dripper).collect();
+
+        _addWithdrawalQueueLiquidity();
+    }
+
+    /// @dev Adds WETH to the withdrawal queue if there is a funding shortfall.
+    /// This assumes 1 WETH equal 1 OETH.
+    function _addWithdrawalQueueLiquidity()
+        internal
+        returns (uint256 addedClaimable)
+    {
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+
+        // Check if the claimable WETH is less than the queued amount
+        uint256 queueShortfall = queue.queued - queue.claimable;
+
+        // No need to get WETH balance if all the withdrawal requests are claimable
+        if (queueShortfall > 0) {
```

Comment:
⚪ Can we change this to short-circuit if?

Reply:
done